### ISSUE TITLE
Custom sort for only once column 

### ISSUE THREAD

**Issue by ucbdiaz2015** [carbon-design-system/carbon#2531]
## Feature request:
I'm looking for a way to add manual sort to only one column in *DataTable*. 

Currently, setting custom sort overrides sorting for all columns. For my use case, however, I only need to override functionality for one column.

One of my columns holds links and the table doesn't seem to know how to handle that, so I'd like to modify sorting for just this column and leave the rest as is since they work just fine.

Using
```
"carbon-components": "^9.8.0",
"carbon-components-react": "6.46.2"
```

**Comment by dakahn** [comment]
Closing this as resolved 🤠

**Comment by 3dfoster** [comment]
> Hi 👋 thank you entering this issue! Just one question - Have you tried setting `false` to [`isSortable` prop in `<TableHeader>`](https://github.com/IBM/carbon-components-react/blob/v6.96.0/src/components/DataTable/TableHeader.js#L120-L123)? Something like:
> 
> ```js
> <TableHeader {...getHeaderProps({ header })} isSortable={false}>
>   {header.header}
> </TableHeader>
> ```

Would this prevent the other columns from using the default sort method?

**Comment by asudoh** [comment]
`<TableHeader>` represents a header for a single column if it answers your question.

**Comment by 3dfoster** [comment]
> `<TableHeader>` represents a header for a single column if it answers your question.

Does setting a TableHeader's `isSortable` to false prevent sorting completely?

**Comment by asudoh** [comment]
> Does setting a TableHeader's isSortable to false prevent sorting completely?

Only for the specific header column IIUC.

**Comment by 3dfoster** [comment]
Being able to supply custom sort for single/specific column(s) would be very helpful to me. The current implementation has a custom sort function being passed to the DataTable, and that sort is applied to all columns. It sounds like the only way to prevent custom sort from being applied to a column is to disable sorting entirely, and that is limiting.

Currently I have to build logic into my custom sort function to detect the header.key for the column it is coming from, and then use a switch statement to sort based on the column. Since We are responsible for rendering the DataTable anyways, it seems like it would be intuitive to pass a custom sort function to any TableHead(s) that needs custom sort.

Can we re-open this issue, or should I create a new issue for this feature request?

**Comment by pramodgslab** [comment]
@3dfoster How to pass custom sort to the datatable ? Any ref ?

**Comment by albertmir** [comment]
I second the request of @3dfoster . It would be really nice to have this feature.

Edited: Found that this was implemented to [carbon-components-svelte](https://github.com/IBM/carbon-components-svelte/pull/352) that offers just this. Can we have the same features?
